### ISSUE TITLE
Apply blog recommendations for solving certain bugs

### ISSUE THREAD

**Issue by SergioQuijanoRey** [SergioQuijanoRey/TFG#72]
I've read a few blogs with some indications on how to implement online triplet loss:

- [First one](https://omoindrot.github.io/triplet-loss#offline-and-online-triplet-mining)
- [Second One](https://towardsdatascience.com/triplet-loss-advanced-intro-49a07b7d8905#%7E:text=This%20is%20due%20to%20the,pairs%20are%20at%20that%20moment)

Follow them as a last try to fix problems with training

**Comment by SergioQuijanoRey** [comment]
Avoid vector space collapsing: [other blog](https://quaterion.qdrant.tech/tutorials/triplet_loss_trick.html)

**Comment by SergioQuijanoRey** [comment]
[This snippet](https://github.com/lyakaap/NetVLAD-pytorch/blob/master/hard_triplet_loss.py) implements Batch hard triplet loss for the first blog, but in Pytorch instead of TensorFlow
